### ISSUE TITLE
KDE Display Issue on Arch Linux

### ISSUE THREAD

**Issue by hammad-awan** [neutrinolabs/xrdp#3181]
### xrdp version

0.10.1-1

### Detailed xrdp version, build options
```
xrdp 0.10.0
  A Remote Desktop Protocol Server.
  Copyright (C) 2004-2024 Jay Sorg, Neutrino Labs, and all contributors.
  See https://github.com/neutrinolabs/xrdp for more information.

  Configure options:
      --prefix=/usr
      --sysconfdir=/etc
      --libexecdir=/usr/lib
      --localstatedir=/var
      --sbindir=/usr/bin
      --with-systemdsystemunitdir=/usr/lib/systemd/system
      --enable-jpeg
      --enable-tjpeg
      --enable-fuse
      --enable-fdkaac
      --enable-opus
      --enable-rfxcodec
      --enable-mp3lame
      --enable-pixman
      --enable-painter
      --enable-vsock
      --enable-ipv6
      --enable-pam-config=arch
      --enable-rdpsndaudin
      --with-imlib2
      CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/home/anonymous/xrdp/src=/usr/src/debug/xrdp -flto=auto
      LDFLAGS=-Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now          -Wl,-z,pack-relative-relocs -flto=auto
      CXXFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/home/anonymous/xrdp/src=/usr/src/debug/xrdp -flto=auto

  Compiled with OpenSSL 3.3.1 4 Jun 2024
```

### Operating system & version

Arch Linux

### Installation method

git clone & make install

### Which backend do you use?

xorgxrdp

### What desktop environment do you use?

KDE

### Environment xrdp running on

Hyper-V VM

### What's your client?

Microsoft RDP client

### Area(s) with issue?

Graphic glitches

### Steps to reproduce

Successfully connect and authenticate using xrdp and a widescreen monitor with a maximum resolution of 5120x1440.

### ✔️ Expected Behavior

To see the desktop.

### ❌ Actual Behavior

I have a Samsung Odyssey G9 monitor with maximum resolution of 5120x1440.

I am able to authenticate and login using xrdp.

However, the screen is only partially updated. I can see the KDE spin icon in a small area of the screen (which keeps spinning) and a small subsection of the bottom left corner where task bar icons are, but most of it is just the default Microsoft blue RDP screen.

### Anything else?

/var/log/xdrp.log
```
[2024-07-27T10:51:37.373-0400] [INFO ] Socket 14: connection accepted from [::ffff:192.168.1.5]:58215
[2024-07-27T10:51:37.377-0400] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[2024-07-27T10:51:37.379-0400] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[2024-07-27T10:51:37.382-0400] [INFO ] Security protocol: configured [SSL|RDP], requested [SSL|HYBRID|HYBRID_EX|RDP], selected [SSL]
[2024-07-27T10:51:37.422-0400] [ERROR] SSL_read: I/O error
[2024-07-27T10:51:37.424-0400] [ERROR] libxrdp_force_read: header read error
[2024-07-27T10:51:37.426-0400] [ERROR] Processing [ITU-T T.125] Connect-Initial failed
[2024-07-27T10:51:37.427-0400] [ERROR] [MCS Connection Sequence] receive connection request failed
[2024-07-27T10:51:37.428-0400] [ERROR] xrdp_sec_incoming: xrdp_mcs_incoming failed
[2024-07-27T10:51:37.429-0400] [ERROR] xrdp_rdp_incoming: xrdp_sec_incoming failed
[2024-07-27T10:51:37.430-0400] [ERROR] xrdp_process_main_loop: libxrdp_process_incoming failed
[2024-07-27T10:51:37.431-0400] [ERROR] xrdp_iso_send: trans_write_copy_s failed
[2024-07-27T10:51:37.432-0400] [ERROR] Sending [ITU T.125] DisconnectProviderUltimatum failed
[2024-07-27T10:51:38.695-0400] [INFO ] Socket 14: connection accepted from [::ffff:192.168.1.5]:58216
[2024-07-27T10:51:38.697-0400] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[2024-07-27T10:51:38.698-0400] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[2024-07-27T10:51:38.700-0400] [INFO ] Security protocol: configured [SSL|RDP], requested [SSL|HYBRID|HYBRID_EX|RDP], selected [SSL]
[2024-07-27T10:51:38.708-0400] [INFO ] Connected client computer name: ANONYMOUS-PC
[2024-07-27T10:51:38.710-0400] [INFO ] client supports gfx protocol
[2024-07-27T10:51:38.712-0400] [WARN ] Received [MS-RDPBCGR] TS_UD_HEADER type 0xc006 is unknown (ignored)
[2024-07-27T10:51:38.713-0400] [WARN ] Received [MS-RDPBCGR] TS_UD_HEADER type 0xc00a is unknown (ignored)
[2024-07-27T10:51:38.716-0400] [INFO ] xrdp_load_keyboard_layout: Keyboard information sent by the RDP client, keyboard_type:[0x04], keyboard_subtype:[0x00], keylayout:[0x00000409]
[2024-07-27T10:51:38.717-0400] [INFO ] xrdp_load_keyboard_layout: model [] variant [] layout [us] options []
[2024-07-27T10:51:38.718-0400] [INFO ] TLS connection established from [::ffff:192.168.1.5]:58216 TLSv1.3 with cipher TLS_AES_256_GCM_SHA384
[2024-07-27T10:51:38.728-0400] [INFO ] Client supports single-screen resizes by xrdp
[2024-07-27T10:51:38.730-0400] [INFO ] xrdp_caps_process_pointer: client supports new(color) cursor
[2024-07-27T10:51:38.732-0400] [INFO ] xrdp_process_offscreen_bmpcache: support level 1 cache size 10485760 MB cache entries 100
[2024-07-27T10:51:38.733-0400] [INFO ] xrdp_caps_process_codecs: NSCodec(CA8D1BB9-000F-154F-589F-AE2D1A87E2D6), codec id [1], properties len [3]
[2024-07-27T10:51:38.734-0400] [INFO ] xrdp_caps_process_codecs: Image RemoteFX(2744CCD4-9D8A-4E74-803C-0ECBEEA19C54), codec id [5], properties len [49]
[2024-07-27T10:51:38.735-0400] [INFO ] Client Capability: LARGE_POINTER_FLAG_96x96 supported
[2024-07-27T10:51:38.738-0400] [INFO ] Loading keymap file /etc/xrdp/km-00000409.ini
[2024-07-27T10:51:38.740-0400] [WARN ] local keymap file for 0x00000409 found and doesn't match built in keymap, using local keymap file
[2024-07-27T10:51:38.742-0400] [INFO ] Login screen monitor height is 1440 pixels over 336 mm (108 DPI)
[2024-07-27T10:51:38.759-0400] [INFO ] xrdp_egfx_create: error 0 channel_id 1
[2024-07-27T10:51:39.576-0400] [INFO ] xrdp_mm_egfx_caps_advertise:
[2024-07-27T10:51:39.602-0400] [INFO ]   version 0x00080004 flags 0x00000000 (index: 0)
[2024-07-27T10:51:39.616-0400] [INFO ]   version 0x00080105 flags 0x00000000 (index: 1)
[2024-07-27T10:51:39.627-0400] [INFO ]   version 0x000a0002 flags 0x00000000 (index: 2)
[2024-07-27T10:51:39.641-0400] [INFO ]   version 0x000a0200 flags 0x00000000 (index: 3)
[2024-07-27T10:51:39.652-0400] [INFO ]   version 0x000a0301 flags 0x00000000 (index: 4)
[2024-07-27T10:51:39.662-0400] [INFO ]   version 0x000a0400 flags 0x00000000 (index: 5)
[2024-07-27T10:51:39.676-0400] [INFO ]   version 0x000a0502 flags 0x00000000 (index: 6)
[2024-07-27T10:51:39.687-0400] [INFO ]   version 0x000a0600 flags 0x00000000 (index: 7)
[2024-07-27T10:51:39.700-0400] [INFO ]   version 0x000a0701 flags 0x00000000 (index: 8)
[2024-07-27T10:51:39.710-0400] [INFO ]   replying version 0x000a0701 flags 0x00000000
[2024-07-27T10:51:39.720-0400] [INFO ] xrdp_mm_egfx_caps_advertise: xrdp_egfx_send_capsconfirm error 0 best_index 8
[2024-07-27T10:51:39.732-0400] [INFO ] xrdp_egfx_reset_graphics:
[2024-07-27T10:51:39.742-0400] [INFO ] xrdp_egfx_reset_graphics: width 5120 height 1440 monitorcount 1
[2024-07-27T10:51:39.751-0400] [INFO ] xrdp_mm_egfx_caps_advertise: xrdp_egfx_send_reset_graphics error 0 monitorCount 0
[2024-07-27T10:51:39.761-0400] [INFO ] xrdp_mm_egfx_create_surfaces: map surface_id 0 left 0 top 0 width 5120 height 1440
[2024-07-27T10:51:39.771-0400] [INFO ] xrdp_encoder_create: starting gfx rfx pro codec session
[2024-07-27T10:51:39.780-0400] [INFO ] xrdp_mm_egfx_invalidate_wm_screen:
[2024-07-27T10:51:39.790-0400] [INFO ] xrdp_mm_egfx_caps_advertise: egfx created.
[2024-07-27T10:51:47.248-0400] [WARN ] Ignoring obsolete SCP port value '3350'
[2024-07-27T10:51:47.562-0400] [INFO ] xrdp_wm_log_msg: sesman connect ok
[2024-07-27T10:51:47.757-0400] [INFO ] xrdp_wm_log_msg: Performing login request for anonymous
[2024-07-27T10:51:47.656-0400] [INFO ] xrdp_wm_log_msg: login was successful - creating session
[2024-07-27T10:51:48.161-0400] [INFO ] xrdp_wm_log_msg: session is available on display 10 for user anonymous
[2024-07-27T10:51:48.164-0400] [INFO ] xrdp_wm_log_msg: Connecting to session
[2024-07-27T10:51:48.167-0400] [INFO ] loaded module 'libxup.so' ok, interface size 10416, version 4
[2024-07-27T10:51:48.168-0400] [INFO ] lib_mod_connect: connecting via UNIX socket
[2024-07-27T10:51:48.169-0400] [INFO ] lib_mod_log_peer: xrdp_pid=2987 connected to Xorg_pid=3002 Xorg_uid=1000 Xorg_gid=1000 client=[::ffff:192.168.1.5]:58216
[2024-07-27T10:51:48.170-0400] [INFO ] xrdp_wm_log_msg: Connecting to chansrv
[2024-07-27T10:51:48.173-0400] [INFO ] Received memory_allocation_complete command. width: 5120, height: 1472
[2024-07-27T10:51:55.696-0400] [ERROR] xrdp_sec_recv: xrdp_mcs_recv failed
[2024-07-27T10:51:55.698-0400] [ERROR] xrdp_rdp_recv: xrdp_sec_recv failed
[2024-07-27T10:51:55.699-0400] [ERROR] libxrdp_process_data: xrdp_rdp_recv failed
[2024-07-27T10:51:55.700-0400] [ERROR] xrdp_process_data_in: xrdp_process_loop failed
[2024-07-27T10:51:55.701-0400] [ERROR] SSL_write: I/O error
[2024-07-27T10:51:55.703-0400] [ERROR] xrdp_iso_send: trans_write_copy_s failed
[2024-07-27T10:51:55.705-0400] [ERROR] Sending [ITU T.125] DisconnectProviderUltimatum failed
```

**Comment by rowlap** [comment]
With a screen size of 5120x1440 you're likely running into #3083.

**Comment by hammad-awan** [comment]
> With a screen size of 5120x1440 you're likely running into #3083.

Thanks. But what does this mean as far as the xrdp source base on the Arch Linux AUR? I see that some files were changed in librfxcodec but how does that apply to the former?

Regards.

**Comment by rowlap** [comment]
That issue was closed on 2024-07-09, but [Arch AUR hasn't picked it up yet](https://aur.archlinux.org/packages?O=0&K=xrdp). I'm not familiar with AUR release plans or policies, but it's likely to be in the next point release.

**Comment by metalefty** [comment]
We'll release v0.10.1 with the fix within a week. Hang tight until a release and AUR maintainer updates the package to v0.10.1.

**Comment by hammad-awan** [comment]
> We'll release v0.10.1 with the fix within a week. Hang tight until a release and AUR maintainer updates the package to v0.10.1.

Haha. thanks for the update. I couldn't wait so I just took the code for the fix from the link rowlap provided that was committed in the source base, overwrite those source files locally for the package , and made and installed the package again.

I'm happy to report that it worked!

**Comment by metalefty** [comment]
@hammad-awan We have released xrdp v0.10.1 and the issue should be resolved.

**Comment by metalefty** [comment]
Closing as resolved.
